File: Singleton.dart
class Pointer {
  static Pointer? _p; //Pricate static attribute

  Pointer() {}
  //private cronstructor
  Pointer? _createSingleton() {
    if (_p == null) {
      _p = new Pointer();
    } else {
      print("Object already created");
    }
    return _p;
  }
}

void main(List<String> args) {
  Pointer p = new Pointer();
  p._createSingleton();
  Pointer p1 = new Pointer();
  p1._createSingleton();
}

File: Singleton.java
class Singleton{
    
    private static Singleton singleton; //Static private attribute

    private Singleton(){

    }
    private Singleton createSingleton(){
        //this.singleton = ( this.singleton == null ) ?  new Singleton() : null;
        if( singleton == null ){
            singleton = new Singleton();
        }else{
            System.out.println("Already created");
        }
        return singleton;
    }
    public static void main(String[] args) {
        Singleton s1 = new Singleton();
        s1.createSingleton(); 
        Singleton s2 = new Singleton();
        s2.createSingleton();       
    }
}